### ISSUE TITLE
7.2 issues

### ISSUE THREAD

**Issue by barsavii** [AdiAddons/AdiBags#169]
I noticed after the patch today that the quest reward [http://www.wowhead.com/item=147860/empowered-elven-tome](url) was not appearing in AdiBags.  I had to turnoff the addon to consume the item.

Also, I do not get a Head equipment slot section nor are head items displayed when equipment is in one section.

**Comment by ManagerWoW** [comment]
4 people in my guild reporting that their Keystone is not showing up.

**Comment by barsavii** [comment]
Thanks for pointing that out.  My keystone is also not showing up.

**Comment by Guzzlord** [comment]
I too am having an issue where Keystones do not show up in my bags.

**Comment by axxidental** [comment]
Items are also appearing in junk slots, as well as in the stack of empty slots.

**Comment by FlesHBoX** [comment]
This also appears to be affecting visibility of artifact power items that I have received from world quests, as well as recipes.  Finally got around to getting the suramar surf and turf today, and the "turf" portion was not showing up, then after combining them, the completed recipe was not showing up.

Hopefully can get a fix for this one.  I don't know if I can handle not having adibags, lol.

**Comment by talchas** [comment]
I had peddler screwing up on account of item link changes or something, and I'm not sure I've seen any issues since fixing peddler - in getUniqueItemId, replace the local bonuses = ... line with

```lua
    local bonuses = ""
    if split[14] then
       bonuses = split[14]:match("^[^|]*")
    end
```

**Comment by xylotism** [comment]
Can confirm everything said above -- keystone, artifact power, random items not showing up -- some of it is fixed by doing "Sort" then "Reset New" but sometimes that just ends up shuffling the items around without changing the icon shown in Adibags -- I just popped a Defiled Augment Rune without reading the tooltip of what looked like the LW recipe I just picked up.

**Comment by heyner88** [comment]
i think it's time for me to move to another bag addon, seems like this one is not been maintained for a while :(

**Comment by seathasky** [comment]
Would love a fix for this, I cant play wow without this addon anymore, it's because my only bag addon of choice for a long time now.

**Comment by seathasky** [comment]
Would love an update for this for 7.2. This addon completes my WoW experience.

**Comment by barsavii** [comment]
Is there anyone willing to work on this code if the current dev is unable or unwilling?

**Comment by legowxelab2z8** [comment]
itemLinks for Mythic keystones have changed. They no longer use the standard item string and now use a mythic keystone string.  
|cffa335ee|Hkeystone:233:3:1:0:0:0|h[Keystone: Cathedral of Eternal Night]|h|r 
This causes addon.IsValidItemLink in AdiBags\core\Utility.lua to return false for Mythic Keystones.
Adding a special case for Mythic Keystones like the special case for battle pets fixes the issue of Keystones not showing up in AdiBags.

**Comment by Taurolyon** [comment]
I am also experiencing this issue, especially when equipping/unequipping gear.

I recently got a legendary ring, and attempted to equip it (not realizing I already had 2 legendaries equipped). This caused the ring to vanish from my inventory. I disabled Adibags and it was in the vanilla bags.

**Comment by tpittma2** [comment]
legowxelab2z8's comment fixed it for me. This is what I did.

Go into AdiBags\core\Utility.lua and open in notepad++

go to line 152 and replace

`	if type(link) == "string" and (strmatch(link, "battlepet:") or (strmatch(link, 'item:[-:%d]+') and not strmatch(link, 'item:%d+:0:0:0:0:0:0:0:0:0'))) then`

with

`if type(link) == "string" and (strmatch(link, "battlepet:") or strmatch(link, "keystone:") or (strmatch(link, 'item:[-:%d]+') and not strmatch(link, 'item:%d+:0:0:0:0:0:0:0:0:0'))) then`

**Comment by orlith** [comment]
Got a similar issue with food recipe which doesn't show up...

**Comment by seathasky** [comment]
Tpittma2, will this fix all of this issues? I appreciate your support but are there any other issues? I know people were having other items missing (Including the AK26 quest item).

**Comment by SpareSimian** [comment]
The keystone fix looks like the battlepet special case, and there are 4 places in the code where battlepets are treated specially (3 in core/Utility.lua and 1 in modules/FilterOverride.lua). Should there be similar changes in the other 3 places?

**Comment by cj-zeiger** [comment]
@seathasky I would put money on their still being issues. I'm sure Tpittma2 did some on the fly debugging, so there is likely to be more issues. If the dev is not interested to maintaining this any further, we need someone to get deeply similar with the codebase and item API or I'm afraid this addon will not be able to continue.

**Comment by legowxelab2z8** [comment]
@SpareSimian Those special cases for battlepets probably also apply for Mythic Keystones as well.  I updated my pull request to add those.

@seathasky, @cj-zeiger There is likely still an issue that is causing some items to not be shown in adibags.

**Comment by cj-zeiger** [comment]
@legowxelab2z8 Could you also look into why this item was not showing up in anywhere in the bag?
(Its the item that upgrades your weapon's artifact knowledge from 25 to 26).

You only ever have it for a short time, but its pretty important to use. I needed to disable adi bags on all my alts just to use it as it wasn't showing up anywhere in the bags.

http://www.wowhead.com/item=147449/enchanted-elven-tome

**Comment by legowxelab2z8** [comment]
Someone commented that a "workaround" for Mythic Keystones not appearing was to use the item sort, and close and reopen your bags.  I played around with it and found that it doesn't work 100% of the time.  I discovered it only works when the Keystone changes position in your inventory. It then inherits the item information of the item it swapped spots with.  Hopefully patching function containerProto:UpdateContent(bag) in widgets\ContainerFrame.lua to use GetItemInfo(itemId) when GetItemInfo(link) fails allows the slotData cache to correctly update and might resolve the issue with other items as well.

**Comment by legowxelab2z8** [comment]
@cj-zeiger I also had a problem with the Enchanted Elven Tome now showing up in AdiBags, but I no longer have one to do further debugging with. I don't notice anything special about the item that should be causing issues with AdiBags but further debugging is certainly needed if this item is always not shown in AdiBags.

**Comment by SpareSimian** [comment]
If you're having trouble with an item and you know how to get access to it, post the item ID. I'm betting there's a pattern to the missing items. (Is there a way to get it from Wowhead?)

**Comment by SpareSimian** [comment]
Ok, on Wowhead, there's a Links button at top right of an item page, and the link for the Enchanted Elven Tome look like this:

"\124cffffffff\124Hitem:147449::::::::110:::::\124h[Enchanted Elven Tome]\124h\124r"

It has the "item:" pattern so should function, but maybe the other fields are causing problems somewhere.

**Comment by legowxelab2z8** [comment]
[Enchanted Elven Tome](http://www.wowhead.com/item=147449/enchanted-elven-tome) ID 147449 is the quest item. [Empowered Elven Tome](http://www.wowhead.com/item=147860/empowered-elven-tome)  ID 147860 is the item you get for completing the quest. The Empowered Elven Tome seems to be a normal item with nothing special about it that should cause issues.

**Comment by seathasky** [comment]
Looks like the author is still around, his other addons are being updated, lets hope this one does too.

**Comment by xylotism** [comment]
Anybody who's having trouble seeing the AK26 Elven Tome, open up your artifact weapon and hover over the number in the top-left -- if the tooltip says Artifact Knowledge 26 it's already been applied.

I did the quest on six different characters yesterday and on two of them I didn't get or have to use the item, it just applied automatically -- doesn't seem to be related to this addon.

**Comment by cheald** [comment]
I had similar issues with a [Pile of Herbs] not showing up, as well as a stack of [Unbending Potion]. It almost seemed like they were being positioned behind other item slots.

**Comment by wheatbread** [comment]
Vantus Rune: Krosus is also not showing up.

**Comment by seathasky** [comment]
AdiDevs pleaseeeeeeee fix the best bad addon to ever be created!

**Comment by DarkOsa** [comment]
If you experience issues with the addon, follow tpittma2's advice from above, looks like it fixed things for now. Items displaying correctly.

**Comment by legowxelab2z8** [comment]
I have been on a scavenger hunt tracking down and acquiring the items listed as having issues.  [Empowered Elven Tome](http://www.wowhead.com/item=147860/empowered-elven-tome)  both [Recipe: Surf](http://www.wowhead.com/item=141011/recipe-surf) and [Recipe: Turf](http://www.wowhead.com/item=141012/recipe-turf) and also [Unbending Potion](http://www.wowhead.com/item=127845/unbending-potion). They all show up in my bags and display fine.  One interesting thing I noticed was that they are all filtered as Consumable.  

I created a [script](https://gist.github.com/legowxelab2z8/d04215b77806aad0186635e8e0408139) to scan my bags and find the corresponding button in AdiBags and report any items that are missing buttons.  There is a bug with the script when it tries to get buttons for full stacks of items. The function containerProto:FindExistingButton in widgets\ContainerFrame.lua that the script uses to get the AdiBags button will remove any buttons that have a full stack and then return nil, so there are some false positives, but it should report any missing items.  

While playing and getting all of the above items I did not have any instances of missing items.  At this point I believe that there isn't a bug for specific items, but maybe a bug for consumables since all the items reported in this issue are consumables. Please continue to report if you notice item missing from your bags.  
I think this issue might also have something to do with the patch just launching and iirc local item caches get reset with a patch. Combined with the increased server load on patch days perhaps item data not being immediately available was causing some issues that I am no longer experiencing.

TL;DR Despite my efforts I cannot reproduce this bug. Please report if this bug is still happening.

**Comment by seathasky** [comment]
Thanks for the update Lego, you're my hero!
